var/spack/repos/builtin/packages/espresso/package.py

```diff
@@ -91,38 +94,55 @@ def install(self, spec, prefix):
             options.append('--enable-openmp')
 
         if '+scalapack' in spec:
-            options.append('--with-scalapack=yes')
+            scalapack_option = 'intel' if '^intel-mkl' in spec else 'yes'
+            options.append('--with-scalapack={0}'.format(scalapack_option))
 
         if '+elpa' in spec:
-            if spec.satisfies('@:5.4.0'):
```

Comment:
I added the conflict and a comment on top of that.

`espresso@5.4.0` seems to support some form of configuration for an external `elpa`, but they construct the wrong include folder based on the prefix you pass to configure.  I went with a conflict as we are talking about an old version, and this is fixed in `@6.1.0`. The other option would be for `elpa` to provide a symlink from the "wrong" folder searched by `espresso`.